modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/info/item/provider/ObjectEntryInfoItemFormProvider.java

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.web.internal.info.item.provider;
+
+import com.liferay.info.exception.NoSuchFormVariationException;
+import com.liferay.info.field.InfoFieldSet;
+import com.liferay.info.form.InfoForm;
+import com.liferay.info.item.field.reader.InfoItemFieldReaderFieldSetProvider;
+import com.liferay.info.item.provider.InfoItemFormProvider;
+import com.liferay.info.localized.InfoLocalizedValue;
+import com.liferay.object.exception.NoSuchDefinitionException;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.model.ObjectEntry;
+import com.liferay.object.model.ObjectField;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectFieldLocalService;
+import com.liferay.object.web.internal.info.item.ObjectEntryInfoItemFields;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.security.permission.ResourceActionsUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+
+import java.util.Locale;
+import java.util.Set;
+
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Jorge Ferrer
+ */
+@Component(
+	immediate = true, property = Constants.SERVICE_RANKING + ":Integer=10",
+	service = InfoItemFormProvider.class
+)
+public class ObjectEntryInfoItemFormProvider
+	implements InfoItemFormProvider<ObjectEntry> {
+
+	@Override
+	public InfoForm getInfoForm() {
+		try {
+			return _getInfoForm(0);
+		}
+		catch (NoSuchFormVariationException noSuchFormVariationException) {
+			throw new RuntimeException(noSuchFormVariationException);
+		}
+	}
+
+	@Override
+	public InfoForm getInfoForm(ObjectEntry objectEntry) {
+		long objectDefinitionId = objectEntry.getObjectDefinitionId();
+
+		try {
+			return _getInfoForm(objectDefinitionId);
+		}
+		catch (PortalException portalException) {
+			throw new RuntimeException(
```

Comment:
I threw a RuntimeException since it was easier as I was working on the prototype, but maybe think through whether there is a better way

Reply:
I discussed with @gabrielwas, we didn't find a better way for now, let's move on. If necessary, we refactor it afterwards.

Reply:
I'm ok to go with this for now, although I think we should improve it.
We can keep it as a RuntimeException if we don't think client code should deal with it, but it's usually better to create a specific subclass of SystemException in case it's desired to catch it.